lib/Connection.js

```diff
@@ -96,53 +94,38 @@ class Connection {
     debug('Starting a new request', params)
     const request = this.makeRequest(requestParams)
 
-    // listen for the response event
-    // TODO: handle redirects?
-    request.on('response', response => {
-      /* istanbul ignore else */
-      if (ended === false) {
-        ended = true
-        this._openRequests--
+    const onResponse = response => {
+      cleanListeners()
+      this._openRequests--
+      callback(null, response)
+    }
 
-        if (params.asStream === true) {
-          callback(null, response)
-        } else {
-          callback(null, decompressResponse(response))
-        }
-      }
-    })
-
-    // handles request timeout
-    request.on('timeout', () => {
-      /* istanbul ignore else */
-      if (ended === false) {
-        ended = true
-        this._openRequests--
-        request.abort()
-        callback(new TimeoutError('Request timed out', params), null)
-      }
-    })
-
-    // handles request error
-    request.on('error', err => {
-      /* istanbul ignore else */
-      if (ended === false) {
-        ended = true
-        this._openRequests--
-        callback(new ConnectionError(err.message), null)
-      }
-    })
+    const onTimeout = () => {
+      cleanListeners()
+      this._openRequests--
+      request.once('error', () => {}) // we need to catch the request aborted error
+      request.abort()
```

Comment:
If you use `request.destroy()` it might solve the v14 problem and it will also not emit an error event so you will no longer need the empty error handler.

Reply:
Thanks for the review! I'm looking into this, as `request.abort` has been deprecated in newer versions of node.
Very likely I'll address this in a separate pr while working on https://github.com/elastic/elasticsearch-js/issues/1297.